velox/common/memory/MemoryUsageTracker.cpp

```diff
@@ -223,29 +223,31 @@ void MemoryUsageTracker::maybeUpdatePeakBytes(int64_t newPeak) {
 }
 
 void MemoryUsageTracker::checkNonNegativeSizes(const char* errmsg) const {
-  // TODO: make these CHECK failures after making usage tracker thread-safe.
-  if (reservationBytes_ < 0) {
-    LOG_EVERY_N(ERROR, kLogEveryN) << "MEMORY: Negagtive reservation bytes "
-                                   << errmsg << " " << reservationBytes_;
-  }
+  VELOX_CHECK_GE(
+      reservationBytes_,
+      0,
+      "Negative reservationBytes_ {}, {}",
+      reservationBytes_,
+      toString());
 }
 
 std::string MemoryUsageTracker::toString() const {
   std::stringstream out;
-  out << "<tracker used " << (currentBytes() >> 20) << "MB available "
-      << (availableReservation() >> 20) << "MB";
+  out << "<tracker used " << succinctBytes(currentBytes()) << " available "
+      << succinctBytes(availableReservation());
   if (maxMemory() != kMaxMemory) {
-    out << " limit " << (maxMemory() >> 20) << "MB";
+    out << " limit " << succinctBytes(maxMemory());
   }
-  out << " (usedReservationBytes_ " << usedReservationBytes_
-      << "B grantedReservationBytes_ " << grantedReservationBytes_
-      << "B reservationBytes_ " << reservationBytes_
-      << "B minReservationBytes_ " << minReservationBytes_ << "B)";
-  out << " numAllocs_[" << numAllocs_ << "] numFrees_[" << numFrees_
+  out << " (usedReservationBytes_[" << succinctBytes(usedReservationBytes_)
```

Comment:
consider removing Bytes_ suffixes from the error message for readability and adding some punctuation

`used 1000B, available 1023.02KB, limit 1.00GB, reservation [used 1000B, granted 1.00MB...`

Reply:
I actually want to use the exactly class field name to make it clear on debugging. Will change it as we can map to code. Thanks!